lib/ssh/src/ssh_sftpd.erl

```diff
@@ -676,12 +676,7 @@ do_open(ReqId, State0, Path, Flags) ->
 	    ssh_xfer:xf_send_status(State0#state.xf, ReqId,
     				    ?SSH_FX_FAILURE, "File is a directory");
 	false ->
-	    AbsPath = case Root of
-			  "" ->
-			      Path;
-			  _ ->
-			      relate_file_name(Path, State0)  
-		      end,
+	    AbsPath = relate_file_name(Path, State0),
```

Comment:
Indeed. Changed the code to check the directory with the correct path. I also added two test cases. The error is was compensated by ssh_xfer:encode_erlang_status(Error) for Vsn=6. For Vsn =< 5 incorrect error code was returned before the fix.

I rebased the code on the current maint branch, but left two commits unsquashed as they address different problems.